src/shared/tomoyo-util.c

```diff
@@ -0,0 +1,41 @@
+/***
+  This file is part of systemd.
+
+  Copyright 2017 Shawn Landden
+
+  systemd is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation; either version 2.1 of the License, or
+  (at your option) any later version.
+
+  systemd is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with systemd; If not, see <http://www.gnu.org/licenses/>.
+***/
+
+#include <unistd.h>
+
+#include "tomoyo-util.h"
+
+static const char version_path[] = "/sys/kernel/security/tomoyo/version";
+
+bool mac_tomoyo_use(void) {
+        static int cached_use = -1;
+
+        if (cached_use < 0) {
+                int r;
+
+                r = access(version_path, F_OK);
+                if (r < 0) {
+                        cached_use = false;
+                } else {
+                        cached_use = true;
+                }
+        }
```

Comment:
What about `cached_use = access(version_path, F_OK) >= 0`? We don't try to make the code as compact as possible, but that'd replace 11 lines with 2. Note that we don't use braces for single line blocks.